topdown/uuid.go

```diff
@@ -31,6 +32,32 @@ func builtinUUIDRFC4122(bctx BuiltinContext, operands []*ast.Term, iter func(*as
 	return iter(result)
 }
 
+func builtinUUIDParse(_ BuiltinContext, operands []*ast.Term, iter func(term *ast.Term) error) error {
+	str, err := builtins.StringOperand(operands[0].Value, 1)
+	if err != nil {
+		return err
+	}
+
+	parsed, err := uuid.Parse(string(str))
+	if err != nil {
+		return nil
+	}
+
+	result := ast.NewObject()
+	for prop, val := range parsed {
+		switch val := val.(type) {
+		case string:
+			result.Insert(ast.StringTerm(prop), ast.StringTerm(val))
+		case int:
+			result.Insert(ast.StringTerm(prop), ast.IntNumberTerm(val))
+		case int64:
+			result.Insert(ast.StringTerm(prop), ast.NumberTerm(int64ToJSONNumber(val)))
```

Comment:
Would we need a default case here?

Reply:
Used `ast.InterfaceToValue` instead as Stephan suggested. But yes it would prob be best to have a default case if I had kept it as is.